Kudu.Services/ServiceHookHandlers/FetchHandler.cs

```diff
@@ -476,6 +498,13 @@ public static void PerformBackgroundDeployment(DeploymentInfo deployInfo, IEnvir
                     backgroundTrace.Dispose();
                 }
             });
+
+            if (waitForTempDeploymentCreation)
+            {
+                // Wait until the temp deployment is created before returning
+                var timeout = TimeSpan.FromSeconds(5);
+                await Task.WhenAny(tcs.Task, Task.Delay(timeout));
```

Comment:
Task.Delay will have an orphan timer in happy.   It will be moot if using auto/manual reset event.   If timeout what to do? Nothing if this is simply best effort to reduce (but not eliminate) the race temp deployment issue.